main/helper.js

```diff
@@ -0,0 +1,245 @@
+  console.log('Burn ERC20 txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+
+  return txHash;
+};
+const withdrawERC20 = async (client, tokenAddress, txHash) => {
+  const erc20RootToken = client.erc20(tokenAddress, true);
+  const result = await erc20RootToken.withdrawExit(txHash);
+  const txHashExit = await result.getTransactionHash();
+  console.log('Exit ERC20 txHash = ', txHashExit);
+  const txReceipt = await result.getReceipt();
+};
+
+// --------------------------------------------------------------------------------------------------------------------------------------------- //
+
+async function calculateBlockNum(timeInterval) {
+  const provider = new ethers.providers.EtherscanProvider('goerli');
+  const currentBlock = await provider.getBlockNumber();
+  const blockTime = 15; // ETH block time is 15 seconds
+
+  //Block number 5 mins ago
+  const block = currentBlock - timeInterval / blockTime;
+
+  return [block, currentBlock];
+}
+
+// Know payBacks
+
+async function txnHistory(address, block5min, currentBlock) {
+  const provider = new ethers.providers.EtherscanProvider('goerli');
+  // Get all txs for address since 5 mins
+  let history = await provider.getHistory(address, block5min, currentBlock);
+  return history;
+}
+
+async function knowPayBacks(
+  myAddress,
+  specificAddress,
+  block5min,
+  currentBlock
+) {
+  const history = await txnHistory(myAddress, block5min, currentBlock);
+  let i = 0;
+  let payBack = [];
+  while (i < history.length) {
+    if (
+      history[i].to == myAddress &&
+      history[i].from != specificAddress &&
+      history[i].value.toString() >= 200000000000000
+    ) {
+      payBack.push({
+        sender: history[i].from,
+        amount: history[i].value.toString(),
+      });
+    }
+    i++;
+  }
+  return payBack;
+}
+
+// async function caller() {
+//     const pays = await knowPayBacks("0x9b52aa46AfaED4E9E5F576d19D369C65F9f3ea58", "0xdd160613122C9b3ceb2a2709123e3020CaDa2546")
+//     console.log(pays)
+// }
+// caller()
+
+// --------------------------------------------------------------------------------------------------------------------------------------------- //
+
+// Know ERC20 payBacks
+
+const erc20ABI = [
+  'event Approval(address indexed src, address indexed guy, uint256 wad)',
+  'event LogNote(bytes4 indexed sig, address indexed usr, bytes32 indexed arg1, bytes32 indexed arg2, bytes data) anonymous',
+  'event Transfer(address indexed src, address indexed dst, uint256 wad)',
+  'function decimals() view returns (uint8)',
+  'function approve(address usr, uint256 wad) returns (bool)',
+  'function transfer(address dst, uint256 wad) returns (bool)',
+];
+
+async function erc20TxnHistory(contractAddress, block5min) {
+  const contractPointer = new ethers.Contract(
+    contractAddress,
+    erc20ABI,
+    ropstenProvider
+  );
+
+  function arrayFilter(transferTxn) {
+    return transferTxn['blockNumber'] >= block5min;
+  }
```

Comment:
We can use inline functions as well, instead of creating function for everything

Reply:
https://github.com/Rushanksavant/Bridging/blob/2e5edd7a88b00d5d8600d661c7fff58f73827f54/main/helper.js#L114
The `.filter()` method takes in a function argument to filter the array, hence I had to define this as a function.